uaclient/gpg.py

```diff
@@ -19,21 +19,16 @@ def export_gpg_key_from_keyring(
         the source_keyring_file.
     """
     export_cmd = [
-        "gpg",
-        "--output",
-        destination_keyfile,
-        "--yes",
-        "--no-auto-check-trustdb",
-        "--no-options",
-        "--no-default-keyring",
+        "apt-key",
         "--keyring",
         source_keyring_file,
-        "--export",
+        "export",
         key_id,
     ]
+    add_ring_cmd = ["apt-key", "--keyring", destination_keyfile, "add", "-"]
```

Comment:
I just tested something and noticed that `apt-key add` will not overwrite a keyring file, but add the key to it if it doesn't exist yet. You won't get duplicates, but if you call it with key foo once, and key bar after, the target file will contain both keys. Should we guard against this, and is it safe to always remove the destination file before calling this command?